core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala

```diff
@@ -253,6 +263,38 @@ class ResourceProfile(
   }
 }
 
+/**
+ * Resource profile which only contains task resources, can be used for stage level task schedule
+ * when dynamic allocation is disabled, tasks will be scheduled to executors with default resource
+ * profile based on task resources described by this task resource profile.
+ * And when dynamic allocation is enabled, will require new executors for this profile base on
+ * default build-in executor resources and assign tasks by resource profile id.
+ *
+ * @param taskResources Resource requests for tasks. Mapped from the resource
+ *                      name (e.g., cores, memory, CPU) to its specific request.
+ */
+@Evolving
+@Since("3.4.0")
+class TaskResourceProfile (
```

Comment:
currently we return it from the build() method so it would have to be public, else we would have to return it as just ResourceProfile there but keep it this internally

Reply:
Doesn't the build() return `ResourceProfile` now? Though a `TaskResourceProfile` is created, the return type is still `ResourceProfile`.

Reply:
So we don't want custom to directly create a `TaskResourceProfile` here? If that's the case, I can make it private.